### PR TITLE
Use Botan for PKCS signature formatting

### PR DESCRIPTION
**Pull request by randombit**
Completing #50. This probably shouldn't be merged quite yet - it relies on an unmerged Botan PR (https://github.com/randombit/botan/pull/1051), still waiting on CI there, but it should be merged to master today or tomorrow.

But, otherwise ready for review.

### CODE REVIEW

**Comment by randombit** [comment]
Necessary PR to Botan has been merged to master and CI is green now - ready for review.

**Comment by ronaldtse** [comment]
Thanks @randombit ! Would it be possible to help add some CMocka tests around these so we don't run into the issue last time?

@dewyatt would you have time to review this? Thanks!

**Comment by randombit** [comment]
@ronaldtse I think the test from #90 already covers this change pretty well, but I will extend it to test all the available hashes.

File: src/lib/rsa.c [hunk 1]
```diff
@@ -174,6 +136,77 @@ pgp_rsa_encrypt_pkcs1(uint8_t *out,
    return retval;
 }
 
+static void rnp_hash_to_botan_pkcs1_padding(char padding_name[], size_t len_padding_name,
+                                            const char* hash_name)
+{
+        // rnp uses SHAx Botan uses SHA-x
+        if(strcmp(hash_name, "SHA1") == 0)
+        {
+                strcpy(padding_name, "EMSA-PKCS1-v1_5(Raw,SHA-1)");
```
Comment (dewyatt) [inline src/lib/rsa.c]: We should use strncpy on all these with `len_padding_name - 1` as the size:
```
strncpy(padding_name, "EMSA-PKCS1-v1_5(Raw,SHA-1)", len_padding_name - 1);
```

Comment (randombit) [inline src/lib/rsa.c]: Good call, fixed in 9c9aa564

**Comment by randombit** [comment]
I extended the test a bit to cover more hashes, also fixed a crashing bug (segfault) which occurred while trying to validate a signature with an unknown hash id - the return value of `pgp_hash_any` wasn't being checked.

**Pull request merged**